### ISSUE TITLE
Android : autoGrantPermissions must also be applied for a full reset or fast reset

### ISSUE THREAD

**Issue by mayureshshirodkar** [appium/appium#7810]
## The problem

Grant permissions if set to true are only applied during the first installation.
On reseting the app via fastReset or driver.resetApp(), the app permission grant set the first time gets cleared.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.6.3
* Desktop OS/version used to run Appium: Mac 10.11.6
* Node.js version (unless using Appium.app|exe): v7.0.0
* Mobile platform/version under test: Android 6.0.1
* Real device or emulator/simulator: Moto G4 Real device
* Appium CLI or Appium.app|exe: Appium CLI

## Details

Describe in more detail the problem you have been experiencing, if necessary.

## Link to Appium logs

https://gist.github.com/mayureshshirodkar/3b0adbbe3fe4b9788db6b54652856565

## Code To Reproduce Issue [ Good To Have ]

		androidCapabilities.setCapability("fullReset", true);
                driver.resetApp()

**Comment by egorvas** [comment]
+1

**Comment by imurchie** [comment]
Can you try the latest in 1.6.4 beta? It should have version 1.13.0 of `appium-android-driver`

**Comment by egorvas** [comment]
Same of this version. Also I found that autoGrantPermissions dosen't work in case when I install application with "app" capability.

**Comment by imurchie** [comment]
Can you provide the logs for a run where it does not work. As a link to a [gist](https://gist.github.com).

**Comment by egorvas** [comment]
I found that in my case location isn't set in case of crash Appium Settings application. 
This is logs of the crash https://gist.github.com/egorvas/e7e556d1612a502be438ad82715b8c99

**Comment by mykola-mokhnach** [comment]
@egorvas The problem is, that settings application tries to get access to location services, but does not have necessary permissions to do so. And thus simply crashes.
It might be just enough to execute 

```
adb shell 'pm grant io.appium.settings android.permission.ACCESS_FINE_LOCATION; pm grant io.appium.settings android.permission.ACCESS_COARSE_LOCATION'
```

after this app is installed. In the worst case it will be necessary to update settings app code to properly advertise the necessary permissions, like it is required in Android 6.0+ (now it works in compatibility mode, which is going to be deprecated).

**Comment by egorvas** [comment]
@mykola-mokhnach In my case Im not able to do that, because I using selenium grid server to run my tests. And Im not able to perform any adb commands on the server side...

**Comment by SrinivasanTarget** [comment]
+1 @mykola-mokhnach I'd faced this crash consistently in Android 7 a week back. I think we should fix it or make necessary changes at our end.

**Comment by mayureshshirodkar** [comment]
The app permission do work for me, on On reseting the app via fastReset or driver.resetApp().

**Comment by mykola-mokhnach** [comment]
@egorvas The fix for location mocking on Android 6+ has been recently published in 1.6.4-beta. Please try it and report if any problems are found there

**Comment by egorvas** [comment]
@mykola-mokhnach I found issue with method  adb.installOrUpgrade(settingsApkPath, SETTINGS_HELPER_PKG_ID);
For some reasons settings app not installing to the device if this application not installed. Please check it, it can cause lots of problems for beta users .
